### PR TITLE
refactor: Dry `handleAuthData` for safer code maintenance in the future

### PR DESCRIPTION
**Pull request by Moumouls**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/parse-server/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/parse-server/blob/alpha/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

## Issue

Duplicated code related to an old GHSA on auth system. A quick dry refactor.

Closes: NONE

## Approach

A small DRY refactor for easiest, safer futur code maintenance on auth.

## Tasks

- [x] Add tests
- [x] Add changes to documentation (guides, repository pages, code comments)
- [x] Add [security check](https://github.com/parse-community/parse-server/blob/master/CONTRIBUTING.md#security-checks)
- [x] Add new Parse Error codes to Parse JS SDK

### CODE REVIEW

File: spec/ParseUser.spec.js [hunk 1]
```diff
@@ -3350,7 +3350,7 @@ describe('Parse.User testing', () => {
 
   it('should not allow updates to emailVerified', done => {
     const emailAdapter = {
-      sendVerificationEmail: () => {},
+      sendVerificationEmail: () => { },
```
Comment (mtrezza) [inline spec/ParseUser.spec.js]: Could you revert these? We usually have no space between empty curly brackets, but I think we are missing a lint rule for this

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: spec/ParseUser.spec.js (hunk 1 above)
Comment (mtrezza) [inline spec/ParseUser.spec.js]:
```suggestion
      sendVerificationEmail: () => {},
```

File: spec/ParseUser.spec.js [hunk 2]
```diff
@@ -3386,7 +3386,7 @@ describe('Parse.User testing', () => {
 
   it('should not retrieve hidden fields on GET users/me (#3432)', done => {
     const emailAdapter = {
-      sendVerificationEmail: () => {},
+      sendVerificationEmail: () => { },
```
Comment (mtrezza) [inline spec/ParseUser.spec.js]:
```suggestion
      sendVerificationEmail: () => {},
```

File: spec/ParseUser.spec.js [hunk 3]
```diff
@@ -3429,7 +3429,7 @@ describe('Parse.User testing', () => {
 
   it('should not retrieve hidden fields on GET users/id (#3432)', done => {
     const emailAdapter = {
-      sendVerificationEmail: () => {},
+      sendVerificationEmail: () => { },
```
Comment (mtrezza) [inline spec/ParseUser.spec.js]:
```suggestion
      sendVerificationEmail: () => {},
```

File: spec/ParseUser.spec.js [hunk 4]
```diff
@@ -3477,7 +3477,7 @@ describe('Parse.User testing', () => {
 
   it('should not retrieve hidden fields on login (#3432)', done => {
     const emailAdapter = {
-      sendVerificationEmail: () => {},
+      sendVerificationEmail: () => { },
```
Comment (mtrezza) [inline spec/ParseUser.spec.js]:
```suggestion
      sendVerificationEmail: () => {},
```

File: spec/ParseUser.spec.js [hunk 5]
```diff
@@ -3521,7 +3521,7 @@ describe('Parse.User testing', () => {
 
   it('should not allow updates to hidden fields', async () => {
     const emailAdapter = {
-      sendVerificationEmail: () => {},
+      sendVerificationEmail: () => { },
```
Comment (mtrezza) [inline spec/ParseUser.spec.js]:
```suggestion
      sendVerificationEmail: () => {},
```

File: spec/ParseUser.spec.js [hunk 6]
```diff
@@ -3546,7 +3546,7 @@ describe('Parse.User testing', () => {
 
   it('should allow updates to fields with maintenanceKey', async () => {
     const emailAdapter = {
-      sendVerificationEmail: () => {},
+      sendVerificationEmail: () => { },
```
Comment (mtrezza) [inline spec/ParseUser.spec.js]:
```suggestion
      sendVerificationEmail: () => {},
```

File: src/RestWrite.js [hunk 7]
```diff
@@ -523,10 +523,15 @@ RestWrite.prototype.handleAuthData = async function (authData) {
   const r = await Auth.findUsersWithAuthData(this.config, authData);
   const results = this.filteredObjectsByACL(r);
 
-  if (results.length > 1) {
+  const userId = this.getUserId();
+  const userResult = results[0];
+
```
Comment (mtrezza) [inline src/RestWrite.js]:
```suggestion
```

File: src/RestWrite.js [hunk 8]
```diff
@@ -1311,7 +1309,7 @@ RestWrite.prototype.handleInstallation = function () {
           throw new Parse.Error(
             132,
             'Must specify installationId when deviceToken ' +
-              'matches multiple Installation objects'
+            'matches multiple Installation objects'
```
Comment (mtrezza) [inline src/RestWrite.js]:
```suggestion
              'matches multiple Installation objects'
```

File: spec/ParseUser.spec.js [hunk 9]
```diff
@@ -3576,8 +3576,8 @@ describe('Parse.User testing', () => {
         expect(e.code).toBe(Parse.Error.OBJECT_NOT_FOUND);
         expect(
           e.message === 'Invalid username/password.' ||
-            e.message ===
-              'Your account is locked due to multiple failed login attempts. Please try again after 1 minute(s)'
+          e.message ===
+          'Your account is locked due to multiple failed login attempts. Please try again after 1 minute(s)'
```
Comment (mtrezza) [inline spec/ParseUser.spec.js]:
```suggestion
            e.message ===
              'Your account is locked due to multiple failed login attempts. Please try again after 1 minute(s)'
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: spec/AuthenticationAdaptersV2.spec.js [hunk 10]
```diff
@@ -860,7 +887,7 @@ describe('Auth Adapter features', () => {
       auth: { challengeAdapter: throwInChallengeAdapter },
     });
     let logger = require('../lib/logger').logger;
-    spyOn(logger, 'error').and.callFake(() => {});
+    spyOn(logger, 'error').and.callFake(() => { });
```
Comment (mtrezza) [inline spec/AuthenticationAdaptersV2.spec.js]:
```suggestion
    spyOn(logger, 'error').and.callFake(() => {});
```

File: spec/AuthenticationAdaptersV2.spec.js [hunk 11]
```diff
@@ -885,7 +912,7 @@ describe('Auth Adapter features', () => {
 
     await reconfigureServer({ auth: { modernAdapter: throwInSetup } });
     logger = require('../lib/logger').logger;
-    spyOn(logger, 'error').and.callFake(() => {});
+    spyOn(logger, 'error').and.callFake(() => { });
```
Comment (mtrezza) [inline spec/AuthenticationAdaptersV2.spec.js]:
```suggestion
    spyOn(logger, 'error').and.callFake(() => {});
```

File: spec/AuthenticationAdaptersV2.spec.js [hunk 12]
```diff
@@ -907,7 +934,7 @@ describe('Auth Adapter features', () => {
 
     await reconfigureServer({ auth: { modernAdapter: throwInUpdate } });
     logger = require('../lib/logger').logger;
-    spyOn(logger, 'error').and.callFake(() => {});
+    spyOn(logger, 'error').and.callFake(() => { });
```
Comment (mtrezza) [inline spec/AuthenticationAdaptersV2.spec.js]:
```suggestion
    spyOn(logger, 'error').and.callFake(() => {});
```

**Review by mtrezza — CHANGES_REQUESTED** [review]

File: spec/ParseUser.spec.js (hunk 1 above)
Comment (mtrezza) [reply spec/ParseUser.spec.js]: Did some investigation and it seems there is currently no lint rule to control this; so we won't bother neither.

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.0.0-alpha.30](https://github.com/parse-community/parse-server/releases/tag/7.0.0-alpha.30)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0-alpha.1](https://github.com/parse-community/parse-server/releases/tag/7.1.0-alpha.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0-beta.1](https://github.com/parse-community/parse-server/releases/tag/7.1.0-beta.1)

**Comment by parseplatformorg** [comment]
🎉 This change has been released in version [7.1.0](https://github.com/parse-community/parse-server/releases/tag/7.1.0)

**Pull request merged**